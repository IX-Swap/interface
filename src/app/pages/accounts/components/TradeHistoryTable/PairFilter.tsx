import { PairSelect } from 'app/pages/accounts/components/PairSelect/PairSelect'
import { SearchQueryFilter } from 'components/SearchQueryFilter/SearchQueryFilter'
import React from 'react'

export const PairFilter = () => {
  return (
    <SearchQueryFilter name='pair'>
      {({ value, onChange }) => (
        <PairSelect
          style={{ minWidth: 150 }}
          value={value}
          label='Pair'
          onChange={event => {
            onChange(event.target.value)
          }}
        />
      )}
    </SearchQueryFilter>
  )
}
