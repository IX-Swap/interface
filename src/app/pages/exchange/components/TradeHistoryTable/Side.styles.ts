import makeStyles from '@mui/styles/makeStyles'
import { OrderSide } from 'types/order'

export const useStyles = makeStyles(theme => ({
  sideColor: {
    color: (props: any) =>
      props.side === OrderSide.BID
        ? theme.palette.success.main
        : theme.palette.error.main
  }
}))
