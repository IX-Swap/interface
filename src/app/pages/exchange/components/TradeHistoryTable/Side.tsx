import { Typography } from '@material-ui/core'
import { useStyles } from 'app/pages/exchange/components/TradeHistoryTable/Side.styles'
import { getOrderSideName } from 'helpers/strings'
import React from 'react'
import { OrderSide } from 'types/order'

export interface SideProps {
  side: OrderSide
}

export const Side = ({ side }: SideProps) => {
  const { sideColor } = useStyles({ side })
  return <Typography className={sideColor}>{getOrderSideName(side)}</Typography>
}
