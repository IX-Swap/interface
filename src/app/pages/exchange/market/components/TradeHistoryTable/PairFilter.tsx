import { PairSelect } from 'app/pages/exchange/market/components/PairSelect/PairSelect'
import { SearchQueryFilter } from 'components/SearchQueryFilter/SearchQueryFilter'
import React from 'react'

export const PairFilter = () => {
  return (
    <SearchQueryFilter name='pair'>
      {({ value, onChange }) => (
        <PairSelect
          style={{ minWidth: 150 }}
          value={value}
          onChange={event => {
            console.log({ value, event })
            onChange(event.target.value as string)
          }}
        />
      )}
    </SearchQueryFilter>
  )
}
