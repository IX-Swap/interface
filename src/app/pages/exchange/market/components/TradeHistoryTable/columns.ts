import { formatDistance } from 'date-fns'
import { formatAmount } from 'helpers/numbers'
import { renderSide } from 'helpers/rendering'
import { TableColumn } from 'types/util'

const renderDate = (date: string) =>
  formatDistance(new Date(date), new Date(), { addSuffix: true })

export const columns: Array<TableColumn<any>> = [
  {
    key: 'date',
    label: 'Date',
    render: renderDate
  },
  {
    key: 'pair',
    label: 'Pair'
  },
  {
    key: 'name',
    label: 'Name'
  },
  {
    key: 'side',
    label: 'Side',
    render: renderSide
  },
  {
    key: 'type',
    label: 'Type'
  },
  {
    key: 'amount',
    label: 'Invested Amount',
    headAlign: 'right',
    align: 'right',
    render: formatAmount
  },
  {
    key: 'price',
    label: 'Unit Price',
    headAlign: 'right',
    align: 'right',
    render: formatAmount
  },
  {
    key: 'total',
    label: 'Total Amount',
    headAlign: 'right',
    align: 'right',
    render: formatAmount
  }
]
