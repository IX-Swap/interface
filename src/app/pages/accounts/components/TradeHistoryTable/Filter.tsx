import { Grid } from '@mui/material'
import { PairFilter } from 'app/pages/accounts/components/TradeHistoryTable/PairFilter'
import React from 'react'
import { BaseFilters } from 'app/components/BaseFilters/BaseFilters'

export const Filters = () => {
  return (
    <BaseFilters>
      <Grid item>
        <PairFilter />
      </Grid>
    </BaseFilters>
  )
}
