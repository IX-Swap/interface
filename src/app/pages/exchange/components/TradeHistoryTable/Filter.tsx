import { Box, Grid, Typography } from '@material-ui/core'
import { SearchFilter } from 'app/components/SearchFilter'
import { DateFilter } from 'app/pages/admin/components/AssignedVirtualAccountsTable/DateFilter'
import { PairFilter } from 'app/pages/exchange/components/TradeHistoryTable/PairFilter'
import React from 'react'

export const Filters = () => {
  return (
    <Grid
      container
      justify='space-between'
      style={{ paddingLeft: 24, paddingRight: 24 }}
    >
      <Grid item xs={12} md={6}>
        <Box width={300}>
          <SearchFilter
            fullWidth
            placeholder='Search'
            inputAdornmentPosition='end'
          />
        </Box>
      </Grid>
      <Grid item xs={12} md={6}>
        <Grid
          container
          direction='row'
          justify='flex-end'
          spacing={2}
          alignItems='center'
        >
          <Grid item>
            <Typography>Date:</Typography>
          </Grid>
          <Grid item>
            <DateFilter name='fromDate' label='From' />
          </Grid>
          <Grid item>
            <DateFilter name='toDate' label='To' />
          </Grid>
          <Grid item>
            <PairFilter />
          </Grid>
        </Grid>
      </Grid>
    </Grid>
  )
}
