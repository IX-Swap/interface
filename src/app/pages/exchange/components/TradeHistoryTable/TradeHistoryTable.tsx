import React from 'react'
import { TableView } from 'components/TableWithPagination/TableView'
import { Grid } from '@mui/material'
import { useQueryFilter } from 'hooks/filters/useQueryFilter'
import { columns } from 'app/pages/exchange/components/TradeHistoryTable/columns'
import { Filters } from 'app/pages/exchange/components/TradeHistoryTable/Filter'
import { useAuth } from 'hooks/auth/useAuth'
import { getIdFromObj } from 'helpers/strings'
import { exchange as exchangeUrl } from 'config/apiURL'
import { exchange as exchangeQueryKeys } from 'config/queryKeys'

export interface TradeHistory {
  date: string
  pair: string
  name: string
  side: 'BUY' | 'SELL'
  type: string
  investedAmount: number
  unitPrice: number
  totalAmount: number
}

export const TradeHistoryTable = () => {
  const { user } = useAuth()
  const userId = getIdFromObj(user)
  const { getFilterValue } = useQueryFilter()

  const filter = {
    search: getFilterValue('search'),
    to: getFilterValue('toDate'),
    from: getFilterValue('fromDate'),
    pair: getFilterValue('pair'),
    orderType: 'PAST'
  }

  return (
    <Grid container direction='column' spacing={2}>
      <Grid item style={{ maxHeight: 70 }}>
        <Filters />
      </Grid>
      <Grid item>
        <TableView<TradeHistory>
          uri={exchangeUrl.userOrders(userId)}
          name={exchangeQueryKeys.userOrders(userId)}
          columns={columns}
          filter={filter}
          paperProps={{ variant: 'elevation', elevation: 0 }}
        />
      </Grid>
    </Grid>
  )
}
