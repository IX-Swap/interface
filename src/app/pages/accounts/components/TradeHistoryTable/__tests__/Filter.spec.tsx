import React from 'react'
import { render } from 'test-utils'
import { Filters } from 'app/pages/accounts/components/TradeHistoryTable/Filter'

describe('Filters', () => {
  afterEach(async () => {
    jest.clearAllMocks()
  })

  it('should match snapshot', () => {
    const { container } = render(<Filters />)

    expect(container).toMatchSnapshot()
  })
})
