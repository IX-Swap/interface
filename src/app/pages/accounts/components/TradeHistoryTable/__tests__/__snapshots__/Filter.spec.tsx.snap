// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filters should match snapshot 1`] = `
<div>
  <div
    class="MuiGrid-root MuiGrid-container makeStyles-wrapper-1 css-1xzqqmo-MuiGrid-root"
  >
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-2 css-7g05lj-MuiGrid-root"
    >
      <div
        class="makeStyles-searchWrapper-2 MuiBox-root css-0"
      >
        <div
          class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-adornedEnd css-1c4a1fp-MuiInputBase-root-MuiOutlinedInput-root"
        >
          <input
            class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd css-1vvkud8-MuiInputBase-input-MuiOutlinedInput-input"
            placeholder="Search"
            type="text"
            value=""
          />
          <div
            class="MuiInputAdornment-root MuiInputAdornment-positionEnd css-1laqsz7-MuiInputAdornment-root"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-colorDisabled MuiSvgIcon-fontSizeMedium css-fuvw14-MuiSvgIcon-root"
              data-testid="SearchIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
              />
            </svg>
          </div>
          <fieldset
            aria-hidden="true"
            class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="css-nnbavb"
            >
              <span
                class="notranslate"
              >
                â€‹
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
    <div
      class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-lg-10 makeStyles-filterWrapper-3 css-1hlpjpi-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 makeStyles-childrenWRapper-4 css-ofrcno-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-root-5 css-112rbl4-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-fhk4wr-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-1"
              id="mui-1-label"
            >
              From
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-jbw6h8-MuiInputBase-root-MuiOutlinedInput-root"
              style="width: 155px;"
            >
              <input
                aria-invalid="false"
                aria-label="Choose date"
                aria-readonly="true"
                class="MuiOutlinedInput-input MuiInputBase-input css-17cvodc-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-1"
                readonly=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    From
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <div
            class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root makeStyles-root-5 css-112rbl4-MuiFormControl-root-MuiTextField-root"
          >
            <label
              class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-fhk4wr-MuiFormLabel-root-MuiInputLabel-root"
              data-shrink="true"
              for="mui-3"
              id="mui-3-label"
            >
              To
            </label>
            <div
              class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-jbw6h8-MuiInputBase-root-MuiOutlinedInput-root"
              style="width: 155px;"
            >
              <input
                aria-invalid="false"
                aria-label="Choose date"
                aria-readonly="true"
                class="MuiOutlinedInput-input MuiInputBase-input css-17cvodc-MuiInputBase-input-MuiOutlinedInput-input"
                id="mui-3"
                readonly=""
                type="text"
                value=""
              />
              <fieldset
                aria-hidden="true"
                class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
              >
                <legend
                  class="css-173wfxe"
                >
                  <span>
                    To
                  </span>
                </legend>
              </fieldset>
            </div>
          </div>
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        />
      </div>
    </div>
  </div>
</div>
`;
